src/main/java/org/spongepowered/common/mixin/core/world/storage/SaveHandlerMixin.java

```diff
@@ -70,9 +77,45 @@
 
     @Shadow @Final private File worldDirectory;
 
+    @Shadow protected abstract void setSessionLock();
+
     @Nullable private Exception impl$capturedException;
     // player join stuff
     @Nullable private Path impl$file;
+    private Set<File> impl$directoriesToCreate = new HashSet<>();
+
+    @Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Ljava/io/File;mkdirs()Z", remap = false))
+    private boolean impl$mkdirs(File dir) {
+        if (WorldManager.NO_FILE_CREATION) {
+            impl$directoriesToCreate.add(dir);
```

Comment:
this.

Reply:
That basically means "if the `SerializationBehavior` is ever switched from `NONE` to something else, create these directories before saving anything." I'll change the name to something more descriptive.

Reply:
I feeel silly for commenting on this much later on... but he means turn this line into `this.impl$directoriesToCreate.add(dir);`